Shared/UserAgentBuilder.swift

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+
+public struct UserAgentBuilder {
+    struct UAVersions {
+        let safariVersion: String
+        let webkitVersion: String
+    }
+    
+    struct OSVersionMap {
+        let majorVersion: Int
+        let minorVersion: Int
+        let uaVersions: UAVersions
+    }
+    
+    public init() {}
+    
+    public func build(appendSafariVersion: Bool = false) -> String {
+        let uaVersions = webkitVersion
+        var ua = "Mozilla/5.0 (\(cpuInfo)) AppleWebKit/\(uaVersions.webkitVersion) (KHTML, like Gecko) Mobile/\(kernelVersion)"
+        
+        if appendSafariVersion {
+            ua += " Safari/\(uaVersions.safariVersion)"
+        }
+        
+        return ua
+    }
+    
+    var cpuInfo: String {
+        let os = ProcessInfo().operatingSystemVersion
+        let osVersion = "\(os.majorVersion)_\(os.minorVersion)"
+        
+        var currentDevice = UIDevice.current.model
+        // Only use first part of device name(so "iPod Touch" becomes "iPod")
+        if let deviceNameFirstPart = currentDevice.split(separator: " ").first {
+            currentDevice = String(deviceNameFirstPart)
+        }
+        
+        let platform = UIDevice.current.userInterfaceIdiom == .pad ? "OS" : "iPhone OS"
+        
+        return "\(currentDevice); CPU \(platform) \(osVersion) like Mac OS X"
+    }
+    
+    var webkitVersion: UAVersions {
+        let _12_0 = OSVersionMap(majorVersion: 12, minorVersion: 0,
+                              uaVersions: UAVersions(safariVersion: "605.1",
+                                                     webkitVersion: "605.1.15"))
```

Comment:
Are these going to be important to manually update you think?

Reply:
We have to.
We will see when it breaks thanks to the unit test

Chromium does it manually too I think https://chromium.googlesource.com/chromium/src.git/+/master/ios/web/common/user_agent.mm#46